packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -993,6 +998,15 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.updateDocumentDirtyState(newState);
     }
 
+    private readonly onOp = (op: ISequencedDocumentMessage) => {
```

Comment:
Please add some comments on why this function is the way it is (i.e. we need to allow for delay-loading, but op processing is synchronous, thus the algorithm).

I'd add (at least for now, not sure if I want to keep it in production) a Boolean that tracks if op processing is paused, mostly to validate that setPaused() has synchronous.
I.e.

        assert(!this.paused);
        const rebaseP = this.pendingStateManager.rebaseAt(op.sequenceNumber);
        if (rebaseP) {
            this.paused = true;
            this.scheduleManager.setPaused(true);
            // eslint-disable-next-line @typescript-eslint/no-floating-promises
            rebaseP.then(() => {
                    this.paused = false;
                    this.scheduleManager.setPaused(false));
            }
        }